client/core/core.go

```diff
@@ -2554,6 +2558,61 @@ func handleTradeSuspensionMsg(c *Core, dc *dexConnection, msg *msgjson.Message)
 	return nil
 }
 
+// handleTradeResumptionMsg is called when a trade resumption notification is received.
+func handleTradeResumptionMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error {
+	var rs msgjson.TradeResumption
+	err := msg.Unmarshal(&rs)
+	if err != nil {
+		return fmt.Errorf("trade resumption unmarshal error: %v", err)
+	}
+
+	dc.pendingResumesMtx.Lock()
+	defer dc.pendingResumesMtx.Unlock()
+
+	// Ensure the provided market exists for the dex.
+	dc.marketMtx.RLock()
+	mkt, ok := dc.marketMap[rs.MarketID]
+	dc.marketMtx.RUnlock()
+	if !ok {
+		return fmt.Errorf("no market found with ID %s", rs.MarketID)
+	}
+
+	// Ensure the market is suspended.
+	if !mkt.Suspended {
+		return fmt.Errorf("market %s for dex %s is not suspended",
+			rs.MarketID, dc.acct.host)
+	}
```

Comment:
It might end up being fair game for the operator to send a suspension message and immediately schedule a resume, but this is good for now.